magpie/api/management/user/user_utils.py

```diff
@@ -872,6 +872,11 @@ def check_user_info(user_name=None, email=None, password=None, group_name=None,
         ax.verify_param(user_name, matches=True, param_name="user_name", param_compare=ax.PARAM_REGEX,
                         http_error=HTTPBadRequest,
                         msg_on_fail=s.Users_CheckInfo_UserNameValue_BadRequestResponseSchema.description)
+        supplemental_regex = get_constant("MAGPIE_SUPPLEMENTAL_USERNAME_REGEX", raise_missing=False)
+        if supplemental_regex:
+            ax.verify_param(user_name, matches=True, param_name="user_name", param_compare=supplemental_regex,
+                            http_error=HTTPBadRequest,
```

Comment:
> And then, simply pass user_name_regex to the existing ax.verify_param check just above.

The idea here is that there this extra regex acts as an _additional_ check as well as ax.PARAM_REGEX not instead of it.

Presumably, the restrictions made by ax.PARAM_REGEX are there for a reason. We don't want to completely ignore that reason by allowing the user to set `MAGPIE_SUPPLEMENTAL_USERNAME_REGEX='^.*$'` (or a similar permissive regex) and then the user name can be anything.

If I'm wrong and the ax.PARAM_REGEX check isn't useful, then let me know and I can do as you suggest.

> Careful also about the mssing raise_not_set parameter that could be problematic if the settings was uncommented but unassigned.

Sounds good

Reply:
Ok. To ensure that at least 1 `[a-zA-Z0-9]` character is used, `PARAM_REGEX` remains valid combined with the extra regex.